stew/conio.nim

```diff
@@ -0,0 +1,337 @@
+      # ReadConsole() is not working for such types of STDIN.
+      if setConsoleCP(CP_UTF8) == 0'i32:
+        return err(ioLastError())
+
+      # Allocating buffer with size equal to `maxBytes` + len(CRLF)
+      var buffer = newString(maxBytes + 2)
+      let bytesToRead = uint32(len(buffer))
+      var bytesRead: uint32
+      let rres = readFile(hConsoleInput, cast[pointer](addr buffer[0]),
+                          bytesToRead, bytesRead, nil)
+      if rres == 0:
+        let errCode = ioLastError()
+        discard setConsoleCP(prevInputCP)
+        return err(errCode)
+
+      if setConsoleCP(prevInputCP) == 0'i32:
+        return err(ioLastError())
+
+      # Truncate additional bytes from buffer.
+      buffer.setLen(int(min(bytesRead, uint32(maxBytes))))
+
+      # Trim CR/CRLF from buffer.
+      if len(buffer) > 0:
+        if buffer[^1] == char(0x0A):
+          if len(buffer) > 1:
+            if buffer[^2] == char(0x0D):
+              buffer.setLen(len(buffer) - 2)
+            else:
+              buffer.setLen(len(buffer) - 1)
+          else:
+            buffer.setLen(len(buffer) - 1)
+        elif buffer[^1] == char(0x0D):
+          buffer.setLen(len(buffer) - 1)
+      ok(buffer)
+    else:
+      let prevMode =
+        block:
+          var mode: uint32
+          let res = getConsoleMode(hConsoleInput, mode)
+          if res == 0:
+            return err(ioLastError())
+          mode
+
+      var newMode = prevMode or ENABLE_PROCESSED_INPUT
+      newMode = newMode and not(ENABLE_ECHO_INPUT)
+
+      # Change console CodePage to allow UTF-8 strings input.
+      if setConsoleCP(CP_UTF8) == 0'i32:
+        return err(ioLastError())
+
+      # Disable local echo output.
+      let mres = setConsoleMode(hConsoleInput, newMode)
+      if mres == 0:
+        let errCode = ioLastError()
+        discard setConsoleCP(prevInputCP)
+        return err(errCode)
+
+      # Allocating buffer with size equal to `maxBytes` + len(CRLF)
+      var buffer = newSeq[Utf16Char](maxBytes + 2)
+      let charsToRead = uint32(len(buffer))
+      var charsRead: uint32
+      let rres = readConsole(hConsoleInput, cast[pointer](addr buffer[0]),
+                             charsToRead, charsRead, nil)
+      if rres == 0'i32:
+        let errCode = ioLastError()
+        discard setConsoleMode(hConsoleInput, prevMode)
+        discard setConsoleCP(prevInputCP)
+        return err(errCode)
+
+      # Restore local echo output.
+      if setConsoleMode(hConsoleInput, prevMode) == 0'i32:
+        let errCode = ioLastError()
+        discard setConsoleCP(prevInputCP)
+        return err(errCode)
+
+      # Restore previous console CodePage.
+      if setConsoleCP(prevInputCP) == 0'i32:
+        return err(ioLastError())
+
+      # Truncate additional bytes from buffer.
+      buffer.setLen(int(min(charsRead, uint32(maxBytes))))
+      # Truncate CRLF in result wide string.
+      if len(buffer) > 0:
```

Comment:
Ditto (`strutils.stripLineEnd`)

Reply:
This works on `int16` chunks, not on chars/bytes.